### PR TITLE
Fix Python2 nested with

### PR DESCRIPTION
**Pull request by paradoxxxzero**
I noticed the tests were failing yesterday so I tried to fix it.
It makes python 2 `with` print nested as the snapshot indicates.
I'm still new to prettier development, so please review :)

### CODE REVIEW

File: src/printer/index.js [hunk 1]
```diff
@@ -984,9 +984,18 @@ function genericPrint(path, options, print) {
...
         : [printWithItem(path, print)];
+      if (options.pythonVersion === "2") {
+        return items.reverse().reduce((parts, item, i) => {
+          const withGroup = group(concat(["with", line, item, ":"]));
+          if (i === items.length - 1) {
+            return concat([withGroup, parts]);
+          }
+          return indent(concat([hardline, withGroup, parts]));
+        }, indent(concat([hardline, printBody(path, print)])));
+      }
```
Comment (j-f1) [inline src/printer/index.js]: Can you add a test that shows this new behavior?

**Comment by paradoxxxzero** [comment]
Unless I'm missing something, there is already one, it's the python 2
version of python_with, the one that was failling.

**Comment by j-f1** [comment]
As far as I can see, there [aren’t any failing tests](https://travis-ci.org/prettier/plugin-python/builds/344878421?utm_source=github_status&utm_medium=notification) right now.

**Comment by paradoxxxzero** [comment]
Ah, so it seems it's an architectural problem.
On my machine with a fresh clone:
```shell
$ git clone https://github.com/prettier/plugin-python.git
$ cd plugin-python
$ yarn install
$ yarn test
 PASS   lint  src/printer/tokens.js
 PASS   lint  src/_util-from-prettier.js
 PASS   lint  ./jest.test.config.js
 PASS   lint  tests_config/raw-serializer.js
 PASS   lint  tests/python_set/jsfmt.spec.js
 PASS   lint  tests/python_list/jsfmt.spec.js
 PASS   lint  src/index.js
 PASS   lint  tests_config/run_spec.js
 PASS   lint  tests/python_dict/jsfmt.spec.js
 PASS   lint  tests/python_lambdas/jsfmt.spec.js
 PASS   lint  tests/python_long/jsfmt.spec.js
 PASS   lint  src/parser.js
 PASS   lint  tests/python_import/jsfmt.spec.js
 PASS   lint  ./jest.eslint.config.js
 PASS   lint  tests/python_if/jsfmt.spec.js
 PASS   lint  tests/python_list_comp/jsfmt.spec.js
 PASS   lint  tests/python_try/jsfmt.spec.js
 PASS   lint  tests/python_not_in/jsfmt.spec.js
 PASS   lint  tests/python_nested_if/jsfmt.spec.js
 PASS   lint  tests/python_tuple/jsfmt.spec.js
 PASS   lint  tests/python_pass/jsfmt.spec.js
 PASS   lint  tests/python_return/jsfmt.spec.js
 PASS   lint  tests/python_args_mixed/jsfmt.spec.js
 PASS   lint  tests/python_args/jsfmt.spec.js
 PASS   lint  tests/python_assert/jsfmt.spec.js
 PASS   lint  tests/python_assign/jsfmt.spec.js
 PASS   lint  tests/python_yield/jsfmt.spec.js
 PASS   lint  tests/python_blank_lines_top_level/jsfmt.spec.js
 PASS   lint  tests/python_blank_lines_indented/jsfmt.spec.js
 PASS   lint  tests/python_aug_assign/jsfmt.spec.js
 PASS   lint  tests/python_set_comp/jsfmt.spec.js
 PASS   lint  tests/python_bool_op/jsfmt.spec.js
 PASS   lint  tests/python_class/jsfmt.spec.js
 PASS   lint  tests/python_starred/jsfmt.spec.js
 PASS   lint  tests/python_strings/jsfmt.spec.js
 PASS   lint  tests/python_call/jsfmt.spec.js
 PASS   lint  tests/python_decorators/jsfmt.spec.js
 PASS   lint  tests/python_comments/jsfmt.spec.js
 PASS   lint  tests/python_compare/jsfmt.spec.js
 PASS   lint  tests/python_subscript/jsfmt.spec.js
 PASS   lint  tests/python_kwargs/jsfmt.spec.js
 PASS   lint  tests/python_default_args/jsfmt.spec.js
 PASS   lint  tests/python_dict_comp/jsfmt.spec.js
 PASS   lint  tests/python_expressions/jsfmt.spec.js
 PASS   lint  tests/python_for/jsfmt.spec.js
 PASS   lint  tests/python_float/jsfmt.spec.js
 PASS   lint  tests/python_for_else/jsfmt.spec.js
 PASS   lint  tests/python_for_kv/jsfmt.spec.js
 PASS   lint  tests/python_generator_expressions/jsfmt.spec.js
 PASS   lint  tests/python_hello/jsfmt.spec.js
 PASS   lint  tests/python_tagged_strings/jsfmt.spec.js
 PASS   lint  tests/python3_args_mixed/jsfmt.spec.js
 PASS   lint  tests/python_while/jsfmt.spec.js
 PASS   lint  tests/python3_f_strings/jsfmt.spec.js
 PASS   lint  tests/python_with/jsfmt.spec.js
 PASS   lint  tests/python3_kwargs_only/jsfmt.spec.js
 PASS   lint  tests/python3_aug_assign/jsfmt.spec.js
 PASS   lint  tests/python3_await/jsfmt.spec.js
 PASS   lint  tests/python3_async_func/jsfmt.spec.js
 PASS   lint  tests/python_yield_from/jsfmt.spec.js
 PASS   lint  tests/python_ann_assign/jsfmt.spec.js
 PASS   lint  tests/python_arg_annotations/jsfmt.spec.js
 PASS   lint  tests/python_elif_multi_body/jsfmt.spec.js
 PASS   lint  src/printer/index.js
 PASS   test  tests/python_dict/jsfmt.spec.js
 PASS   test  tests/python_lambdas/jsfmt.spec.js
 PASS   test  tests/python_import/jsfmt.spec.js
 PASS   test  tests/python_set/jsfmt.spec.js
 PASS   test  tests/python_tuple/jsfmt.spec.js
 PASS   test  tests/python_list/jsfmt.spec.js
 PASS   test  tests/python_long/jsfmt.spec.js
 PASS   test  tests/python_class/jsfmt.spec.js
 PASS   test  tests/python_call/jsfmt.spec.js
 PASS   test  tests/python_bool_op/jsfmt.spec.js
 PASS   test  tests/python_yield/jsfmt.spec.js
 PASS   test  tests/python_comments/jsfmt.spec.js
 PASS   test  tests/python_while/jsfmt.spec.js
 FAIL   test  tests/python_with/jsfmt.spec.js
  ● with.py - python-verify

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot 1.

    - Snapshot
    + Received

    @@ -10,9 +10,7 @@
          index = f.read()

      with open("tmp/index.html"):
          print("great")

    - with A() as X:
    -     with B() as Y:
    -         with C() as Z:
    -             do_something()
    + with A() as X,B() as Y,C() as Z:
    +     do_something()

      at Object.test (tests_config/run_spec.js:40:62)
          at new Promise (<anonymous>)
          at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:160:7)

 › 1 snapshot test failed.
 PASS   test  tests/python_blank_lines_top_level/jsfmt.spec.js
 PASS   test  tests/python_blank_lines_indented/jsfmt.spec.js
 PASS   test  tests/python_tagged_strings/jsfmt.spec.js
 PASS   test  tests/python_try/jsfmt.spec.js
 PASS   test  tests/python_assign/jsfmt.spec.js
 PASS   test  tests/python_subscript/jsfmt.spec.js
 PASS   test  tests/python_assert/jsfmt.spec.js
 PASS   test  tests/python_strings/jsfmt.spec.js
 PASS   test  tests/python_args_mixed/jsfmt.spec.js
 PASS   test  tests/python_starred/jsfmt.spec.js
 PASS   test  tests/python_set_comp/jsfmt.spec.js
 PASS   test  tests/python_return/jsfmt.spec.js
 PASS   test  tests/python_not_in/jsfmt.spec.js
 PASS   test  tests/python_pass/jsfmt.spec.js
 PASS   test  tests/python_nested_if/jsfmt.spec.js
 PASS   test  tests/python_kwargs/jsfmt.spec.js
 PASS   test  tests/python_list_comp/jsfmt.spec.js
 PASS   test  tests/python_generator_expressions/jsfmt.spec.js
 PASS   test  tests/python_if/jsfmt.spec.js
 PASS   test  tests/python_hello/jsfmt.spec.js
 PASS   test  tests/python_for_kv/jsfmt.spec.js
 PASS   test  tests/python_args/jsfmt.spec.js
 PASS   test  tests/python_for/jsfmt.spec.js
 PASS   test  tests/python_for_else/jsfmt.spec.js
 PASS   test  tests/python_compare/jsfmt.spec.js
 PASS   test  tests/python_float/jsfmt.spec.js
 PASS   test  tests/python_dict_comp/jsfmt.spec.js
 PASS   test  tests/python_expressions/jsfmt.spec.js
 PASS   test  tests/python3_async_func/jsfmt.spec.js
 PASS   test  tests/python3_args_mixed/jsfmt.spec.js
 PASS   test  tests/python_default_args/jsfmt.spec.js
 PASS   test  tests/python_arg_annotations/jsfmt.spec.js
 PASS   test  tests/python_ann_assign/jsfmt.spec.js
 PASS   test  tests/python_aug_assign/jsfmt.spec.js
 PASS   test  tests/python3_kwargs_only/jsfmt.spec.js
 PASS   test  tests/python_decorators/jsfmt.spec.js
 PASS   test  tests/python_yield_from/jsfmt.spec.js
 PASS   test  tests/python3_f_strings/jsfmt.spec.js
 PASS   test  tests/python3_await/jsfmt.spec.js
 PASS   test  tests/python3_aug_assign/jsfmt.spec.js
 PASS   test  tests/python_elif_multi_body/jsfmt.spec.js

Summary of all failing tests
 FAIL  tests/python_with/jsfmt.spec.js
  ● with.py - python-verify

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot 1.

    - Snapshot
    + Received

    @@ -10,9 +10,7 @@
          index = f.read()

      with open("tmp/index.html"):
          print("great")

    - with A() as X:
    -     with B() as Y:
    -         with C() as Z:
    -             do_something()
    + with A() as X,B() as Y,C() as Z:
    +     do_something()

      at Object.test (tests_config/run_spec.js:40:62)
          at new Promise (<anonymous>)
          at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:160:7)


Snapshot Summary
 › 1 snapshot test failed in 1 test suite. Inspect your code changes or run with `yarn test -- -u` to update them.

Test Suites: 1 failed, 118 passed, 119 total
Tests:       1 failed, 170 passed, 171 total
Snapshots:   1 failed, 106 passed, 107 total
Time:        13.674s
Ran all test suites in 2 projects.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Maybe the python 2 formating is incorrectly parsed with python 3, I'll investigate...

**Comment by paradoxxxzero** [comment]
Found the culprit, this line: https://github.com/prettier/plugin-python/blob/master/src/parser.js#L25 assumes that the default python version is python 2 which is not the case on archlinux for exemple.

**Comment by paradoxxxzero** [comment]
Follow up in #48

**Pull request closed without merging**